packages/clay-table/src/index.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+import Body, {TableBodyType} from './Body';
+import Cell, {TableCellType} from './Cell';
+import Head, {TableHeadType} from './Head';
+import Row, {TableRowType} from './Row';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * This property vertically align the contents
+	 * inside the table body according a given position.
+	 */
+	bodyVerticalAlignment?: VerticalAlignmentType;
+
+	/**
+	 * Applies a Bordered style on Table.
+	 */
+	bordered?: boolean;
+
+	/**
+	 * This property keeps all the headings on one line.
+	 */
+	headingNoWrap?: boolean;
+
+	/**
+	 * This property vertically align the contents
+	 * inside the table header according a given position.
+	 */
+	headVerticalAlignment?: VerticalAlignmentType;
+
+	/**
+	 * Applies a Hover style on Table.
+	 */
+	hover?: boolean;
+
+	/**
+	 * This property enables keeping everything on one line.
+	 */
+	noWrap?: boolean;
+
+	/**
+	 * Turns the table responsive.
+	 */
+	responsive?: boolean;
+
+	/**
+	 * Defines the responsive sizing.
+	 */
+	responsiveSize?: ResposiveSizeType;
+
+	/**
+	 * Applies a Striped style on Table.
+	 */
+	striped?: boolean;
+
+	/**
+	 * This property vertically align the contents
+	 * inside the table according a given position.
+	 */
+	tableVerticalAlignment?: VerticalAlignmentType;
+}
+
+type ResposiveSizeType = 'lg' | 'md' | 'sm' | 'xl';
+
+type VerticalAlignmentType = 'bottom' | 'middle' | 'top';
+
+const ClayTable: React.FunctionComponent<Props> & {
+	Body: TableBodyType;
+	Cell: TableCellType;
+	Head: TableHeadType;
+	Row: TableRowType;
+} = ({
+	bodyVerticalAlignment,
```

Comment:
I'm just a bit reluctant with this bunch of APIs that we have to set up the classes, this is pretty cool but I'm worried because it raises our maintenance layer above the API's, that's a bit worrisome, I'd try to put it on the balance and not need to add everything we have in `classes`. One thing that can help us with that and add only those that the Lexicon provides and `className` will do the rest.

Reply:
What do you think about it @bryceosterhaus @pat270? Would be a good idea leave some APIs using css utilities on the component?